code/components/extra-natives-five/src/WaterExtraNatives.cpp

```diff
@@ -0,0 +1,148 @@
+	{
+		return false;
+	}
+	bool isInsideMin = (_mm_movemask_ps(_mm_cmple_ps(*entityCoords, bounds.max)) & 7) == 7; //Check every coordinate component to be less than the max and create a mask per component -1 = true, 0 = false
+	bool isInsideMax = (_mm_movemask_ps(_mm_cmple_ps(bounds.min, *entityCoords)) & 7) == 7; //Convert the result to an integer mask, all 3 bits need to be set aka 7.
+
+	return isInsideMin && isInsideMax;
+}
+
+void* g_testForWaterVisualsAddr = 0;
+void* g_testForWaterPhysicsAddr = 0;
+
+void* _TestForUnderWaterVisuals(__m128* entityCoords, bool a2, uint64_t a3) //vanilla version
+{
+	static void* (*fn)(__m128*, bool, uint64_t) = (decltype(fn))g_testForWaterVisualsAddr;
+	return fn(entityCoords, a2, a3);
+}
+
+void* _TestForWaterPhysics(void* a1, __m128* entityCoords, uint64_t a3) //vanilla version
+{
+	static void* (*fn)(void*, __m128*, uint64_t) = (decltype(fn))g_testForWaterPhysicsAddr;
+	return fn(a1, entityCoords, a3);
+}
+bool ShouldVolumeBeDry(__m128* entityCoords)
+{
+	bool result = false;
+	for (const sDepthBound& area : g_dryAreas)
+	{
+		result += IsInBounds(entityCoords, area);
+	};
+	return result;
+}
+
+void* CustomTestForWaterPhysics(void* a1, __m128* entityCoords, uint64_t a3)
+{
+	if (ShouldVolumeBeDry(entityCoords))
+	{
+		return 0;
+	}
+	return _TestForWaterPhysics(a1, entityCoords, a3);
+}
+
+void* CustomTestForUnderwaterVisuals(__m128* entityCoords, bool a2, uint64_t a3)
+{
+	if (ShouldVolumeBeDry(entityCoords))
+	{
+		return 0;
+	}
+	return _TestForUnderWaterVisuals(entityCoords, a2, a3);
+}
+
+static HookFunction initFunction([]()
+{
+	g_testForWaterPhysicsAddr = hook::get_pattern("48 8B C4 55 53 56 57 41 54 41 56 41 57 48 8D A8 ? ? ? ? 48 81 EC ? ? ? ? F3 0F 10 42 ? F3 0F 10 0A F3 0F 10 52 ? 0F 29 70 B8 0F 29 78 A8 44 0F 29 40 ? 45 8A F0");
+	g_testForWaterVisualsAddr = hook::get_pattern("48 8B C4 48 89 58 08 48 89 68 18 48 89 70 20 57 48 81 EC ? ? ? ? 83 48 10 FF");
+
+	//Physics calls
+	hook::call(hook::get_pattern("E8 ? ? ? ? 48 8B CB E8 ? ? ? ? 44 38 3D ? ? ? ? 74 10 8A 83"), CustomTestForWaterPhysics);
+	hook::call(hook::get_pattern("E8 ? ? ? ? EB 07 83 A1 ? ? ? ? ? 48 83 C4 38 C3"), CustomTestForWaterPhysics);
+	hook::call(hook::get_pattern("E8 ? ? ? ? 0F 28 CE 48 8B CF E8 ? ? ? ? 66 89 B7 ? ? ? ? 89 B7 ? ? ? ? 8A 8F ? ? ? ? 3A 8F"), CustomTestForWaterPhysics);
+
+	//Visual calls
+	hook::call(hook::get_pattern("E8 ? ? ? ? 33 F6 84 C0 0F 84 ? ? ? ? F3 0F 10 4F ? F3 0F 10 07 E8"), CustomTestForUnderwaterVisuals);
+
+	fx::ScriptEngine::RegisterNativeHandler("CREATE_DRY_VOLUME", [=](fx::ScriptContext& context)
+	{
+		auto newVolume = new sDepthBound(
+		new float[]{
+		context.GetArgument<float>(0),
+		context.GetArgument<float>(1),
+		context.GetArgument<float>(2),
+		},
+		new float[]{
+		context.GetArgument<float>(3),
+		context.GetArgument<float>(4),
+		context.GetArgument<float>(5),
+		});
+
+		std::vector<sDepthBound>::iterator v = std::find(g_dryAreas.begin(), g_dryAreas.end(), newVolume);
```

Comment:
It seems AABB values needs to be limited, there's some weird side effects. If you're creating a dry volume with too high values (in my case I was testing -9999 to 9999), and your local ped is swimming, after creation this region some big part of the map would work like there's some water region (only physically). Wonder why "ped is swimming" part is very important for repro'ing this bug.
Here's repro video: https://gfycat.com/PlaintiveFaithfulAffenpinscher

Reply:
It seems to be the case that creating a dry volume in an area where your ped is swimming, causes this. I cannot seem to figure out how to force the game to somehow reset the swimming state for now. However, dry volumes are likely created when the client joins since its a pretty static map feature. I don't see any usecase for removing water effects if a player is already in that pool anyways. Unless you think this is problematic enough to stop going into production?